### PR TITLE
Add Icinga 2 component

### PR DESCRIPTION
**Pull request by bobapple**

### CODE REVIEW

**Comment by ivansenic** [comment]
Closes #16

File: model/components/icinga.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+      - GPLv2
+    categories:
+      - agent
+      - collector
+      - visualization
+      - dashboarding
+      - alerting
+    capabilities:
+      aspects:
+        - metrics
```
Comment (ivansenic) [inline model/components/icinga.yml]: Can you add what sub-category of metrics are you fulfilling? Currently we have `metrics [app]`, `metrics [process]` and  `metrics [host]`. 

This way tool would still appear in the generic metric filter, but also when `metrics [host]` is selected (if host is added). I believe this is way better than just metrics. If these 3 metrics sub-categories are not enough for you, you can also suggest new ones adn we can disucss it.

Other than that PR is fine, we can merge it.. Thanks for contrib!

File: model/components/icinga.yml [hunk 2]
```diff
@@ -0,0 +1,31 @@
+---
+components:
+
+  # Icinga 2
+  - id: icinga2
```
Comment (ivansenic) [inline model/components/icinga.yml]: Would it be better to split into several components? Until now we always at least divided agents from collectors/visualization/.. The reason is that by this way you are saying that your agent can be connected to MySql for example, which is not the case I belive. I guess your agent can be connected to your collector and the collector can be connected to then to other tools. 

The best way usually is to divide into really separate components, meaning one component can run separately from the other. For us, this is also important because of the future features we are planning on the landscape website.

File: model/components/icinga.yml (hunk 1 above)
Comment (bobapple) [reply model/components/icinga.yml]: Thanks for the feedback, the metrics aspects available are just fine, I will provide an update.

Regarding the general `categories`: Are there any plans to add a category for availability monitoring?

File: model/components/icinga.yml (hunk 2 above)
Comment (bobapple) [reply model/components/icinga.yml]: I thought about splitting everything up into several component. I would rather split Icinga into separate components, but I thought about the following issues, maybe you can help:

- Does it make sense to split Icinga 2 and Icinga Web 2? The one is the core that runs checks, provides data and stores them into MySQL, PostgreSQL, Graphite and so on. The other is the web interface to display everything. Both of them can be extended with features and modules.
- Icinga 2 runs on the server and also on the agent side. It's the same binary, but depending on it's configuration it can act as a server or agent. I understand the concern. An Icinga agent should not be connected to any database.
- We have even more components, like Icingabeat, Logstash Output for Icinga or several modules for Icinga Web 2 to display data from Graphite or Elasticsearch. Should they be added as separate components as well?
- We don't have logos for all of these components, how should I handle this?

In short, I would add even more components from the Icinga ecosystem, I'm just not sure how to handle it and if it would fit the purpose of this landscape.

Comment (ivansenic) [reply model/components/icinga.yml]: `Does it make sense to split Icinga 2 and Icinga Web 2?`

I would say yes definitely.

`Icinga 2 runs on the server and also on the agent side. It's the same binary, but depending on it's configuration it can act as a server or agent. I understand the concern. An Icinga agent should not be connected to any database.`

This one is tricky. We did not have such situation until now. From point of being same binary for me this is same component, but as you said then if acting as agent it should not be connected to any storage tools. So here I would also suggest to split.

`We have even more components, like Icingabeat, Logstash Output for Icinga or several modules for Icinga Web 2 to display data from Graphite or Elasticsearch. Should they be added as separate components as well?`

I think this is easy. 
* For the modules don't add any new component, just mark that Icinga Web 2 can be connected to Graphite or Elasticsearch. Modules are just the config you start the already existing component 
* For the logstash ouput, I would also say just connect Icinga server to the Logstash. As this is plugin again for the existing component.
* The beats could be considered as new component imo. I guess this can be in the Transformation category?

`We don't have logos for all of these components, how should I handle this?`

Does not matter, use the same logo everywhere, this is fine. Many tools are already handled like this. However, if any component has it's own logo add it.

As I understand we would end up for 4 components:

* Icinga Agent
* Icinga Server
* Icinga Web
* Icingabeats

Just make sure that connectins, aspects and other properties are correctly divided and defined now.

**Thanks again for the effort, I wanted to do this myself long time ago, but it's always way better that somebody knowing the tool complelty does it.**

**Comment by ivansenic** [comment]
Hi @bobapple,

we add a small adaptation to your model and pushed the stuff, it's already on the website.

However, with your model, we found some small bugs on our website. Anyhow we have some questions around Icingabeat

1. Can Icinga Server send data to Logstash/Elasticsearch without the beat?
2. Who is exposing the Icinga API, can this also be the agent? If so we should also connect icingabeat to the agent?
3. We are thinking to keep beat only in the transformations category as it's a pipe between the api and the elk stack and not a real data collector. For us collector is something that receives the data.

Maybe you can check yourself the http://openapm.io and see how is this looking at the moment?

**Comment by bobapple** [comment]
`Can Icinga Server send data to Logstash/Elasticsearch without the beat?`
Well, both is possible, depending on what the user want's to achieve. You can use the built-in `elasticsearch` feature of Icinga 2 Server to send metrics to Elasticsearch. Icingabeat is a little different. It runs standalone somewhere in your network. It connects to the Icinga Server REST API and receives an event stream. This event stream includes all events happening within Icinga Server (State changes of checks, alerts sent to users, downtimes set by users and so an). So Icingabeat is not about metrics, it's about internal events of Icinga. These events are forwarded to Elasticsearch/Logstash by Icingabeat, so you can correlate those events with logs for example. Perhaps, in your landscape it's more like a collector?

`Who is exposing the Icinga API, can this also be the agent? If so we should also connect icingabeat to the agent?``
Technically, the agent _could_ expose the API, but this is not the case in normal setups. The REST API used by Icingabeat is exposed by the Icinga Server.

`We are thinking to keep beat only in the transformations category as it's a pipe between the api and the elk stack and not a real data collector. For us collector is something that receives the data.`
Icingabeat is a beat as all other Elastic Beats. It is written based on the same libraries and is being built with the same mechanisms. From that  point of view, I would say put it to the same categories where `Beats` are. On the other hand, you have `Beats` in general and I'm not sure if Icingabeat, as a single beat, fits at all?

The website looks good to me, thank you for the quick update! Except Icingabeat, I am not sure myself where to put it.

**Comment by ivansenic** [comment]
Hi @bobapple,

Thanks for the clarification, I think I got the full picture finally.

Although Icingabeat is implementing Beats "technology", I would not put it to the same category as elastic beats (currently agent). Agents should not depend on other components to deliver data. And plus as understand Icingabeat are not really delivering metrics.

I am still seeing this as a `Transformation` category as it takes data from a source and pushes it to the other component. That's exactly why we created  `Transformation` category.

However, as they are only sending internal events of Icinga I am actually thinking if we should completely remove it. Yes you can say that alerting or downtimes are all related to APM, but in order to keep the landscape as compact as possible, maybe we don't need it. For other components (like Prometheus) we could add tens of other connectors, plugins, etc, but this would just overflow the landscape. What do you think? Alternatively, we could keep it in the `Transformation` category.

@AlexanderWert would you agree with me?
